src/main/java/com/redhat/devtools/lsp4ij/features/highlight/LSPTargetElementEvaluator.java

```diff
@@ -36,7 +44,13 @@ public class LSPTargetElementEvaluator extends TargetElementEvaluatorEx2 {
             return null;
         }
 
-        // Try to find the word at the caret and return a fake PSI element for it
+        // See if the view provider can provide an element
+        FileViewProvider viewProvider = file.getViewProvider();
```

Comment:
If this file uses a file view provider based on semantic tokens, we can use it to find the semantic token-based element at the specified offset.